apis/v1alpha1/certificate_authority.go

```diff
@@ -42,7 +42,7 @@ type CertificateAuthoritySpec struct {
 	CertificateAuthorityConfiguration *CertificateAuthorityConfiguration `json:"certificateAuthorityConfiguration"`
 	// The type of the certificate authority.
 	// +kubebuilder:validation:Required
-	CertificateAuthorityType *string `json:"certificateAuthorityType"`
+	CertificateAuthorityType *string `json:"type,omitempty"`
```

Comment:
If you want to rename `certificateAuthorityType` into `type` let's also rename the Go field name `CertificateAuthorityType`